packages/taiko-client/prover/proof_submitter/proof_submitter.go

```diff
@@ -134,6 +151,37 @@ func (s *ProofSubmitter) SubmitProof(
 		"tier", proofWithHeader.Tier,
 	)
 
+	// Check if we still need to generate a new proof for that block.
+	proofStatus, err := rpc.GetBlockProofStatus(ctx, s.rpc, proofWithHeader.BlockID, s.proverAddress)
+	if err != nil {
+		return err
+	}
+	if proofStatus.IsSubmitted {
```

Comment:
If there is a proof submitted, but its invalid, shall we still submit the current proof?

Reply:
No, it depends on the handle of receiving `TransitionProved` or `TransitionContested` event.